manticore/core/smtlib/constraints.py

```diff
@@ -73,13 +73,11 @@ def __len__(self):
             return len(self._constraints) + len(self._parent)
         return len(self._constraints)
 
-    def add(self, constraint, check=False):
```

Comment:
Strictly speaking, one is supposed to access constraint sets via `state.constrain`, but in many places our code uses `state.constraints.add`. We should leave this `check` parameter in place just in case any user code relies on it, and update the comment to explain what it actually does.

Reply:
At present, as @feliam pointed out [in another comment](https://github.com/trailofbits/manticore/pull/1668/files/fdb4265b759b55d55958d9800aa4edad50920de3#r409885491), `check` is unused throughout the codebase.  Also, the current bit of code in this method that attempted to do anything with `check` was broken and would not possibly work.

@feliam @ehennenfent, would you guys like me to fix this `check` implementation, or delete the currently unused parameter entirely?

Reply:
@feliam says in Slack:
> It is not tested and it is pretty easy to emulate   in debug situations.  Kill it
> We'll re add it (with it corresponding tests) if we find a clear need in the future